### PR TITLE
Fix slots leaking issue

### PR DESCRIPTION
**Pull request by sideshowdave7**
# What
Fixes slots leaking due to a failed send_reply directly before a send_ready. Also add hostname to the socket IDENTITY for much easier debugging.

# Why
To avoid slot leakage due to `TypeError: Frame 4 ({'value': '<...: ... -...) does not support the buffer interface.`

# Tests
Had to modify the test for checking send_ready's since thats done a little differently now.  Testing that now relies on the worker processes actually starting up.

### CODE REVIEW

File: eventmq/jobmanager.py [hunk 1]
```diff
@@ -75,9 +76,13 @@ def __init__(self, *args, **kwargs):
         """
         super(JobManager, self).__init__(*args, **kwargs)
 
+        setup_logger("eventmq")
+
         #: Define the name of this JobManager instance. Useful to know when
         #: referring to the logs.
-        self.name = kwargs.pop('name', generate_device_name())
+        prefix = (socket.gethostname() + ":").encode('ascii')
```
Comment (com4) [inline eventmq/jobmanager.py]: in 0.4 i just allow the user to define what they want the name to be. Do you think we should just backport that functionality here or do you prefer this (in which case i will 'forwardport' this)?

with the 0.4 method the user would do something like `` emq-jobmanager --name `hostname` ``

Comment (sideshowdave7) [inline eventmq/jobmanager.py]: I think we should nuke this in 0.4 and use our method

Comment (sideshowdave7) [inline eventmq/jobmanager.py]: In fact lets backport that into this if possible.

Comment (com4) [inline eventmq/jobmanager.py]: will do

Comment (sideshowdave7) [inline eventmq/jobmanager.py]: Ok i removed the prefix logic in this PR.

**Comment by com4** [comment]
https://github.com/sideshowdave7/eventmq/pull/1 has the --name argument

**Pull request closed without merging**